docs/v2/adding-new-codegenerated-resource.md

```diff
@@ -0,0 +1,83 @@
+# Adding a new code generated resource to ASO v2
+
+This document discusses how to add a new resource to the ASO v2 code generation configuration. Check out [this PR](TODO) if you'd like to see what the end product looks like.
+
+## What resources can be code generated?
+Any ARM resource can be generated. There are a few ways to determine if a resource is an ARM resource:
+1. If the resource is defined in the [ARM template JSON schema](https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json), or the [auto-generated ARM template JSON schema](https://schema.management.azure.com/schemas/common/autogeneratedResources.json) it is an ARM resource.
+2. If the resource is defined in a `resource-manager` folder in the [Azure REST API specs](https://github.com/Azure/azure-rest-api-specs/tree/main/specification) repo, it is an ARM resource.
+
+If the resource is not in either of the above places and cannot be deployed via an ARM template, then it's not an ARM resource and currently cannot be code generated.
+
+## Determine a resource to add
+There are three key pieces of information required before adding a resource to the code generation configuration file, and each of them can be found in the 
+[ARM template JSON schema](https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json) or the [auto-generated ARM template JSON schema](https://schema.management.azure.com/schemas/common/autogeneratedResources.json). To get started, find the entry for the resource in one of the two templates mentioned. For example, the entry for Azure Network Security Groups looks like this:
+`{ "$ref": "https://schema.management.azure.com/schemas/2020-11-01/Microsoft.Network.json#/resourceDefinitions/networkSecurityGroups" },`
+
+**Note**: In many cases there will be multiple entry's for the same resource, each with a different `api-version`. It is _strongly_ recommended that you use the latest available `api-version`.
+
+### The `name` of the resource
+You usually know this going in. In our example above, the name of the resource is `networkSecurityGroups`.
+
+### The `group` the resource is in 
+This is usually named after the Azure service, for example `microsoft.resources` or `microsoft.documentdb`. In our example entry from above, this is `microsoft.network`.
+
+### The `api-version` of the resource
+This is usually a date, sometimes with a `-preview` suffix. In our example entry from above, this is `2020-11-01`.
```

Comment:
Above it reads

> There are three key pieces of information required

So maybe use a numbered list here to identify the three items

```suggestion
1. The `name` of the resource
   You usually know this going in. In our example above, the name of the resource is `networkSecurityGroups`.

2. The `group` the resource is in 
   This is usually named after the Azure service, for example `microsoft.resources` or `microsoft.documentdb`. In our example entry from above, this is `microsoft.network`.

3. The `api-version` of the resource
   This is usually a date, sometimes with a `-preview` suffix. In our example entry from above, this is `2020-11-01`.
```

Reply:
Done